apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/ActivationType.java

```diff
@@ -0,0 +1,134 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.configuration;
+
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.GlobalTracer;
+import co.elastic.apm.agent.util.PrivilegedActionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.lang.management.ManagementFactory;
+import java.util.List;
+
+public enum ActivationType {
+    K8S_ATTACH, // https://github.com/elastic/apm-mutating-webhook
+    PROGRAMMATIC_SELF_ATTACH, // ElasticApmAttacher.attach();
+    JAVAAGENT_FLAG, // -javaagent:... command-line option used
+    AWS_LAMBDA_LAYER, // Only if installed by using layers (as other metadata already identifies lambda)
+    ENV_ATTACH, // JAVA_TOOL_OPTIONS env var used to specify -javaagent option
+    APM_AGENT_ATTACH_CLI, // 'java -jar apm-agent-attach-cli.jar ...' used
+    FLEET, // Fleet using 'java -jar apm-agent-attach-cli.jar ...' to attach (directly or through webhook)
+
+    AZURE_FUNCTIONS, //Not defined yet how this is identified
+    UNKNOWN;
+
+    private static final Logger logger = LogManager.getLogger(ActivationType.class);
+
+    public String toReferenceString() {
+        return toString().replace('_', '-').toLowerCase();
+    }
+
+    public static ActivationType findType() {
+        ActivationType type = _findType();
+        logger.debug("findType() returning {}",type);
+        return type;
+    }
+
+    public static ActivationType _findType() {
+        // Really just a big `switch` type method
+
+        ElasticApmTracer tracer = GlobalTracer.getTracerImpl();
+        if (tracer == null) {
+            //Not yet finished initializing, or we're in test mode, in which case
+            return UNKNOWN;
+        }
+
+        CoreConfiguration.ActivationMethod method = tracer.getConfig(CoreConfiguration.class).getActivationMethod();
+        logger.debug("CoreConfiguration.ActivationMethod: {}", method);
+        if (!method.equals(CoreConfiguration.ActivationMethod.NONE)){
+            //the activation method was specified by the attacher, so use this as best option
+            switch (method) {
+                case K8S: return K8S_ATTACH;
+                case REMOTE: return APM_AGENT_ATTACH_CLI;
+                case SELF_ATTACH: return PROGRAMMATIC_SELF_ATTACH;
+                case FLEET: return FLEET;
+            }
+        }
+
+        //The next picks up `Container image' lambdas but only if the layer is used, which we
+        //aren't sure is good or bad. We could also check ELASTIC_APM_AWS_LAMBDA_HANDLER
+        //but have deliberately decided not to (for now)
+        String lambdaWrapper = PrivilegedActionUtils.getEnv("AWS_LAMBDA_EXEC_WRAPPER");
+        logger.debug("AWS_LAMBDA_EXEC_WRAPPER: {}", lambdaWrapper);
+        if (null != lambdaWrapper && lambdaWrapper.equals("/opt/elastic-apm-handler")) {
+            return AWS_LAMBDA_LAYER;
+        }
```

Comment:
Since you already added `activation_method` as a config option, wouldn't setting 
```
export ELASTIC_APM_ACTIVATION_METHOD=AWS_LAMBDA_LAYER
``` 
in [`elastic-apm-handler`](https://github.com/elastic/apm-agent-java/blob/c8c8769d91efd2c153b96fb472cb601e2e8c0b7f/elastic-apm-agent/src/main/assembly/elastic-apm-handler) just do?

If it's in order to support existing customers, I would say it doesn't worth the heuristics.

Reply:
yes, the reasoning was lower touchpoints in other areas for a very tiny higher effort here, plus I didn't want to burden the lambda user with yet another env var in their system - in particular one that is entirely non-functional for them (ie keep their cognitive load down, and avoid superfluous questions to us and support ("why is ELASTIC_APM_ACTIVATION_METHOD set and what does it do blah blah blah")

Reply:
OK, I never considered env variables as such.
It's just that this is just too sensitive to `/opt/elastic-apm-handler` exactly.
Why was it decided not to rely on `ELASTIC_APM_AWS_LAMBDA_HANDLER`?

Reply:
I don't see that env var in the docs https://www.elastic.co/guide/en/apm/agent/java/current/aws-lambda.html ?

Reply:
The only one that is relevant to doc here is `AWS_LAMBDA_EXEC_WRAPPER`, which sets [elastic-apm-handler](https://github.com/elastic/apm-agent-java/blob/c8c8769d91efd2c153b96fb472cb601e2e8c0b7f/elastic-apm-agent/src/main/assembly/elastic-apm-handler) as the wrapper script that will be executed during the lambda runtime startup. This script does what we want it to do, for example - setting  configuration options with pre-known values. Those do not require documentation as they are set automatically with known values. I proposed to add your new config option as well:
```
export ELASTIC_APM_ACTIVATION_METHOD=AWS_LAMBDA_LAYER
```
Maybe that's why you thought it causes more installation burden and I didn't- this will not change the installation flow at all.

Reply:
Not the way I see it. Being consistent in what you do in each platform makes much more sense to me. Either way, it is not consistent in this sense already- in fleet you set a config option through the CLI (which makes the most sense and easy to find) and in k8s you set a system properties (as opposed to all others that are set as env variables). 

Speaking on k8s, I think this indeed needs to change. Why appending it to an environment variable if it is a config option that can be set as a separate value? 
I propose making it consistent with the NodeJS and .NET settings - see https://github.com/elastic/apm-mutating-webhook/pull/70